### ISSUE TITLE
normalize(a) for multidimensional arrays

### ISSUE THREAD

**Issue by jw3126** [JuliaLang/julia#34234]
Currently `norm` is definined for arbitrary dimension `Arrays`, while `normalize` only works on vectors,
```julia
arr = collect(reshape(1:6, (2,3)))
norm(arr)      # works
normalize!(arr) # throws
normalize(arr) # throws
```
I find this odd and it bites me every now and then. For instance I often want to normalize before plotting.
Can we have `normalize/normalize!` methods that works on `AbstractArray` (or ducktyped method on `Any`)?

**Comment by stevengj** [comment]
I suspect that this is a relic of the days prior to #27401 when `norm` only worked on 1d arrays.   Now that `norm` works on multi-dimensional arrays, I agree that `normalize` should as well.

Should be an easy change to [this code](https://github.com/JuliaLang/julia/blob/13d1bd4689e2afc12ddf5f48ad22efd13f01782a/stdlib/LinearAlgebra/src/generic.jl#L1584-L1652):

1. Change the type signatures from `AbstractVector` to `AbstractArray`

2. Change [this line](https://github.com/JuliaLang/julia/blob/13d1bd4689e2afc12ddf5f48ad22efd13f01782a/stdlib/LinearAlgebra/src/generic.jl#L1646) from `v[1]` to `first(v)`.  (This is a bug anyway, since `v[1]` is wrong for non 1-based arrays.)

3. Add a few tests.

**Comment by dkarrasch** [comment]
Seems like this was closed in 8f9dd5d.
